src/otx_misp/__init__.py

```diff
@@ -60,32 +68,17 @@ def tag_event(misp, event, tag):
     :param tag: tag to add
     :return: None
     """
-    if not hasattr(misp, '_otx_tags_cache'):
-        misp._otx_tags_cache = misp.get_all_tags()['Tag']
-    for exist_tag in misp._otx_tags_cache:
+    global _otx_tags_cache
+    for exist_tag in _otx_tags_cache:
         if exist_tag['name'] == tag:
             tag_id = exist_tag['id']
-            if 'EventTag' in event:
-                for evt_tag in event['EventTag']:
+            if 'Tag' in event:
```

Comment:
EventTag is certainly wrong, but I believe you must first extract the event dict using key 'Event', and *then* look for the tags dict using key 'Tag'. Here's the first part of a pretty-print of the event object in this function courtesy of the python debugger: 

```
(Pdb) pp(event)
{'Event': {'Attribute': [{'Galaxy': [],
                          'ShadowAttribute': [],
                          'category': 'External analysis',
                          'comment': '',
                          'deleted': False,
                          'disable_correlation': False,
    ...deleted...
           'ShadowAttribute': [],
           'Tag': [{'colour': '#ffffff',
                    'exportable': True,
                    'hide_tag': False,
                    'id': '3',
                    'local': 0,
                    'name': 'foo:white',
   ...deleted...
```